components/shared/richText/RichTextElement.tsx

```diff
@@ -265,7 +249,7 @@ export const RichTextElement: FC<ElementProps> = (props) => {
 
 type RichTextValueProps = Readonly<{
   element: Elements.RichTextElement;
-  value: IPortableTextItem[];
+  value: PortableTextItem[];

```

Comment:
this works, but ideally, the value should be of type `PortableTextObject[]`, because that narrows it down to the top level objects only, which is expected.